.github/workflows/test.yml

```diff
@@ -228,36 +358,68 @@ jobs:
     - name: "Config perftools build"
       working-directory: ".\\perftools\\source"
       run: |
-        cmake -S . -B .\build -DOPENSSL_ROOT_DIR="$env:GITHUB_WORKSPACE\openssl"
+        cmake -S . -B .\build -DOPENSSL_ROOT_DIR="$env:GITHUB_WORKSPACE\openssl" -Drun_add_version_dep:BOOL=OFF ${{ matrix.release.cmakeopts }}
     - name: "Build perftools"
       working-directory: ".\\perftools\\source"
       run: |
          cmake --build .\build --config Release
+    - name: "Print perftools version information"
+      working-directory: ".\\perftools\\source"
+      run: |
+         cmake --build .\build --config Release --target run-version -j 4
+    - name: "Run perftools"
+      working-directory: ".\\perftools\\source"
+      run: |
+         cmake --build .\build --config Release --target run -j 4
 
   macos:
     strategy:
       fail-fast: false
       matrix:
-        openssl-branch:
-          - "openssl-3.0"
-          - "openssl-3.2"
-          - "openssl-3.3"
-          - "openssl-3.4"
-          - "openssl-3.5"
-          - "master"
+        release: [
+          {
+            openssl-branch: "openssl-3.0",
+            cmakeopts: '-D "run_newrawkey_algos=newrawkey;-a;x25519" -D "run_evp_fetch_pqs=evp_fetch;;" -D "run_handshake_pool_size=handshake;;;-o 4"',
+            configopts: 'no-tests',
+          }, {
+            openssl-branch: "openssl-3.2",
+            cmakeopts: '-D "run_newrawkey_algos=newrawkey;-a;x25519" -D "run_evp_fetch_pqs=evp_fetch;;" -D "run_handshake_pool_size=handshake;;;-o 4"',
+            configopts: 'no-apps no-tests',
+          }, {
+            openssl-branch: "openssl-3.3",
+            cmakeopts: '-D "run_newrawkey_algos=newrawkey;-a;x25519" -D "run_evp_fetch_pqs=evp_fetch;;" -D "run_handshake_pool_size=handshake;;;-o 4"',
+            configopts: 'no-apps no-tests',
+          }, {
+            openssl-branch: "openssl-3.4",
+            cmakeopts: '-D "run_newrawkey_algos=newrawkey;-a;x25519" -D "run_evp_fetch_pqs=evp_fetch;;" -D "run_handshake_pool_size=handshake;;;-o 4"',
+            configopts: 'no-apps no-tests',
+          }, {
+            openssl-branch: "openssl-3.5",
+            cmakeopts: '-D "run_handshake_pool_size=handshake;;;-o 4"',
+            configopts: 'no-apps no-tests',
+          }, {
+            openssl-branch: "openssl-3.6",
+            cmakeopts: '-D "run_handshake_pool_size=handshake;;;-o 4"',
+            configopts: 'no-apps no-tests',
+          }, {
+            openssl-branch: "master",
+            cmakeopts: '',
+            configopts: 'no-apps no-tests',
```

Comment:
What's the reason that 3.{5,6} and master skips `-D "run_newrawkey_algos=newrawkey;-a;x25519" -D "run_evp_fetch_pqs=evp_fetch;;" -D "run_handshake_pool_size=handshake;;;-o 4"`?